cmake/modules/HCT.cmake

```diff
@@ -49,6 +55,15 @@ function(add_hlsl_hctgen mode)
                        ${hctdb}
                        ${hctdb_helper})
 
+  get_filename_component(output_extension ${full_output} LAST_EXT)
+
+  if (output_extension MATCHES "\.h|\.cpp|\.inl")
```

Comment:
Should this also check for `.hpp` extensions?

Reply:
We don't have any `.hpp` files, and I think we shouldn't add any...

Reply:
Probably not affected by codegen?

https://github.com/search?q=repo%3Amicrosoft%2FDirectXShaderCompiler%20path%3A*.hpp&type=code

Reply:
Two of those are tests and the third... I wish didn't exist and am now sad. But they shouldn't be impacted here.